docs/Developer-Guide.md

```diff
@@ -219,15 +219,36 @@ rustup target add x86_64-unknown-linux-musl
 sudo ln -s /usr/bin/g++ /bin/musl-g++
 ```
 
+The agent is built with seccomp capability by default.
+Therefore, you **MUST** install `libseccomp` library and set environment variables for
+linking the agent binaries with `libseccomp` library statically before building the agent.
+
+To install `libseccomp` library:
+
+```
+$ $GOPATH/src/github.com/kata-containers/kata-containers/install_libseccomp.sh /tmp/kata-libseccomp /tmp/kata-gperf
+```
+
+> **Note**
+>
+> - You should build `libseccomp` library from sources using the above script to generate the static library for `musl`.
+
+To set environment variables for static link:
+
+```
+$ export LIBSECCOMP_LINK_TYPE=static
+$ export LIBSECCOMP_LIB_PATH=/tmp/kata-libseccomp/lib
+```
+
 To build the agent:
 
 ```
 $ go get -d -u github.com/kata-containers/kata-containers
 $ cd $GOPATH/src/github.com/kata-containers/kata-containers/src/agent && make
 ```
 
-The agent is built with seccomp capability by default.
 If you want to build the agent without the seccomp capability, you need to run `make` with `SECCOMP=no` as follows.
+In addition, if you do not set `LIBSECCOMP_LINK_TYPE` or `LIBSECCOMP_LIB_PATH`, the agent is built with `SECCOMP=no` forcibly.
```

Comment:
This contradicts the "by default" above, and is in my opinion an undesirable behavior.

What about changing it to (and implementing as):

- If you set `LIBSECCOMP_LINK_TYPE` to `static`, `LIBSECCOMP_LIB_PATH` must be set or the build with fail.
- If you set `LIBSECCOMP_LIB_PATH` explicitly, then it must point to a `libseccomp` library built in your build environment, and `LIBSECCOMP_LINK_TYPE` will default to `static` if not set.
- If you set neither variables and you have not set `SECCOMP=no`, then the agent will assume that `libseccomp` is available dynamically, and report an error otherwise.

Reply:
Thank you for the suggestions.

> If you set LIBSECCOMP_LINK_TYPE to static, LIBSECCOMP_LIB_PATH must be set or the build with fail.

I added this to the docs.

> If you set LIBSECCOMP_LIB_PATH explicitly, then it must point to a libseccomp library built in your build environment, and LIBSECCOMP_LINK_TYPE will default to static if not set.

I think this seems to have an impact on the downstream because the [`libseccomp` crate](https://crates.io/crates/libseccomp) sets `LIBSECCOMP_LINK_TYPE` to `dylib` by default if users do not set it explicitly.

> If you set neither variables and you have not set SECCOMP=no, then the agent will assume that libseccomp is available dynamically, and report an error otherwise.

I added the description to the docs.